site/content/blog/kapp-versioned-resource.md

```diff
@@ -0,0 +1,569 @@
+
+
+Let's verify the value of environment variable `MSG_KEY` in the running pod of the deployment `simple-app`.
+
+```bash
+$ kubectl get pods
+NAME                          READY   STATUS    RESTARTS   AGE
+simple-app-5f94df997b-g76d9   1/1     Running   0          25s
+
+$ kubectl exec -it simple-app-5f94df997b-g76d9 sh
+# 
+# echo $MSG_KEY
+hello-kapp
+# 
+```
+The value of environment variable `MSG_KEY` is same as we defined in configmap `simple-config`, so the changes got updated to the deployment without restarting it's pod manually.
+
+Let's update the value of `data.hello_msg` to `hello-tanzu` in configmap and redeploy the app `app` with the updated configmap.
+
+
+```bash
+$ kapp deploy -a app -f app.yaml --diff-changes
+Target cluster 'https://127.0.0.1:33907' (nodes: minikube)
+
+@@ create configmap/simple-config-ver-2 (v1) namespace: default @@
+  ...
+  1,  1   data:
+  2     -   hello_msg: hello-kapp
+      2 +   hello_msg: hello-tanzu
+  3,  3   kind: ConfigMap
+  4,  4   metadata:
+@@ update deployment/simple-app (apps/v1) namespace: default @@
+  ...
+123,123                 key: hello_msg
+124     -               name: simple-config-ver-1
+    124 +               name: simple-config-ver-2
+125,125           image: docker.io/dkalinin/k8s-simple-app:latest
+126,126           name: simple-app
+
+Changes
+
+Namespace  Name                 Kind        Age  Op      Op st.  Wait to    Rs  Ri  
+default    simple-app           Deployment  15h  update  -       reconcile  ok  -  
+^          simple-config-ver-2  ConfigMap   -    create  -       reconcile  -   -  
+
+Op:      1 create, 0 delete, 1 update, 0 noop, 0 exists
+Wait to: 2 reconcile, 0 delete, 0 noop
+
+Continue? [yN]: y
+```
+The new set of resources in our `app`:
+```bash
+$ kapp inspect -a app
+Target cluster 'https://127.0.0.1:33907' (nodes: minikube)
+
+Resources in app 'app'
+
+Namespace  Name                         Kind        Owner    Rs  Ri  Age  
+default    simple-app                   Deployment  kapp     ok  -   15h  
+^          simple-app-5f94df997b        ReplicaSet  cluster  ok  -   15h  
+^          simple-app-6757478ff5        ReplicaSet  cluster  ok  -   15h  
+^          simple-app-6757478ff5-xbq2c  Pod         cluster  ok  -   15h  
+^          simple-config-ver-1          ConfigMap   kapp     ok  -   15h  
+^          simple-config-ver-2          ConfigMap   kapp     ok  -   15h  
+
+Rs: Reconcile state
+Ri: Reconcile information
+
+6 resources
+
+Succeeded
+```
+If you look carefully the new set of resources having:
+1. **two configmaps:** `simple-config-ver-1` with older changes and `simple-config-ver-2` with new changes
+2. **two replicasets:**`simple-app-5f94df997b` with no running pods (older one, kapp deleted it's pod) and `simple-app-6757478ff5` with one running pod and it has the news changes of configmap as well.
```

Comment:
i dont think we should mention replicasets since that's not something that kapp is orchestrating. kapp just updated Deployment. (`older one, kapp deleted it's pod` => kapp only updated Deployment, it did not touch pods)

Reply:
updated